### DISCUSSION TITLE
Show line number

### DISCUSSION THREAD

**Discussion by desprit** [nvim-tree/nvim-tree.lua#1578] [Q&A] [answered]
Is there an option to enable line numbers? Something like in a NerdTree:

```
" enable line numbers
let NERDTreeShowLineNumbers=1
" make sure relative line numbers are used
autocmd FileType nerdtree setlocal relativenumber
```

If I manually do `setlocal relativenumber` in an opened tree it works but I don't know how to make that permanent.

**Comment by kyazdani42** [comment]
`lua require'nvim-tree.view'.View.win_opts.relativenumber = true` after nvim-tree finished setting up. But this wont apply automatically and youll need to setlocal relativenumber to have the change be immediate or toggling the tree :)

**Comment by desprit** [comment]
@kyazdani42 

Ok, thanks. I'll keep using this for now:
```
func! OpenTree()
    :NvimTreeToggle
    :setlocal relativenumber
endfun
nnoremap <F3> :call OpenTree()<CR>
```

Maybe line number support will be part of `nvim-tree` API one day? :)

**Comment by sindrets** [comment]
@desprit There's no need for it to be part of the API. Achieving what you describe here is a pretty common application for auto commands. I would just add an auto command like this:

```vim
au BufWinEnter NvimTree setlocal rnu
```

**Comment by aegatlin** [comment]
I can't get this autocommand to work. I also don't know how to call `lua require'nvim-tree.view'.View.win_opts.relativenumber = true` after nvim-tree setup because there's no `require 'nvim-tree'.setup()` function. Any suggestions?

Variants I've tried:
```
autocmd BufWinEnter NvimTree setlocal relativenumber
autocmd BufWinEnter NvimTree set relativenumber
autocmd FileType NvimTree setlocal relativenumber
autocmd FileType NvimTree set relativenumber
```

I can get it working with wrapper functions around each nvim-tree function call I use, but I would prefer not using that approach.

**Comment by desprit** [comment]
@sindrets 

Has something changed? `au BufWinEnter NvimTree setlocal rnu` no longer works for me.

**Comment by kyazdani42** [comment]
`require'nvim-tree.view'.View.winopts.relativenumber = true`

**Comment by natedub** [comment, marked as answer]
For others stumbling onto this from Google, this can now be configured in the setup call:
```lua
require('nvim-tree').setup({view = {relativenumber = true}})
```
